# This is a BitKeeper generated patch for the following project:
# Project Name: Linux VDR
# This patch format is intended for GNU patch command version 2.5 or higher.
# This patch includes the following deltas:
#	           ChangeSet	1.21    -> 1.22   
#	               eit.c	1.6     -> 1.7    
#
# The following is the BitKeeper ChangeSet Log
# --------------------------------------------
# 02/04/01	aschultz@warp10.net	1.22
# fix segfault in eit.c
# --------------------------------------------
#
diff -Nru a/eit.c b/eit.c
--- a/eit.c	Mon Apr  1 14:38:00 2002
+++ b/eit.c	Mon Apr  1 14:38:00 2002
@@ -559,7 +559,7 @@
      pSubtitle = compactspace(pSubtitle);
      pExtendedDescription = compactspace(pExtendedDescription);
      // Remove superfluous hyphens:
-     if (pExtendedDescription) {
+     if (pExtendedDescription && strlen(pExtendedDescription) >= 3) {
         char *p = pExtendedDescription + 1;
         while (*p) {
               if (*p == '-' && *(p + 1) == ' ' && *(p + 2) && islower(*(p - 1)) && islower(*(p + 2))) {
